### PR TITLE
Events

### PR DESCRIPTION
**Pull request by Kakalanp**
- Adding [big.js](https://www.npmjs.com/package/big-js).
- Adding functions provided by Microverse for the calculator to work.
- Adding event handlers to the calculator buttons.
- Adding a pair of comments for better readability.
- Adding a sorting algorithm that creates buttons from an array and adds class names accordingly.

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,68 +1,86 @@
...
+  const [next, setNext] = useState(null);
+  const [operation, setOperation] = useState(null);
+
+  function clickHandler(e) {
+    const result = calculate({ total, next, operation }, e.target.textContent);
+    setTotal(result.total);
+    setNext(result.next);
+    setOperation(result.operation);
+  }
+
```
Comment (Gambit142) [inline src/components/Calculator.js]: - [x] Nice work done on this project. Your Calculator component should be a class-based component for now. There is nothing wrong with how you have it now, but the project requirements state that you should use _**React Lifecycle methods**._ The only way you can use a lifecycle method is by turning your component into a class-based component. You can read this [article](https://www.w3schools.com/react/react_lifecycle.asp) to have an idea of how to use lifecycle methods. 👍

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,68 +1,86 @@
...
       </div>
-      <div className={styles.calculator__button__grey}>
-        <p>.</p>
+      <div className={styles.calculator__button__grey} onClick={clickHandler} onKeyPress={(e) => { if (e.key === 'Enter') clickHandler(); }} role="button" tabIndex={0}>
+        <p style={{ userSelect: 'none' }}>.</p>
       </div>
-      <div className={styles.calculator__button__orange}>
-        <p>=</p>
+      <div className={styles.calculator__button__orange} onClick={clickHandler} onKeyPress={(e) => { if (e.key === 'Enter') clickHandler(); }} role="button" tabIndex={0}>
+        <p style={{ userSelect: 'none' }}>=</p>
```
Comment (Gambit142) [inline src/components/Calculator.js]: - [x] (Optional) You can put all the symbols and numbers in an array as strings, then map through the array and display every symbol/number dynamically. This way, your code will be dry and there won't be repetition. You can read this [article](http://www.hackingwithreact.com/read/1/13/rendering-an-array-of-data-with-map-and-jsx) as a guide. 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @Kakalanp 👋, 

Good job so far!

### Highlights
- Linters checks passed ✔️ 
- Correct use of logic files ✔️ 

However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_KIndly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider the optional change I suggested.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/components/Calculator.js (hunk 2 above)
Comment (Kakalanp) [reply src/components/Calculator.js]: Thanks for the suggestion! I did an algorithm like that and added the required conditionals for class-naming. 🎉

File: src/components/Calculator.js (hunk 1 above)
Comment (Kakalanp) [reply src/components/Calculator.js]: I completely missed that requirement, I just did the refactoring, thanks! 😁

**Review by bimbolabuari — APPROVED** [review]
Hi @Kakalanp, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**